js/src/components/contact-information/phone-number-card.js

```diff
@@ -0,0 +1,144 @@
+/**
+ * External dependencies
+ */
+import { getCountryCallingCode } from 'libphonenumber-js';
+import { __ } from '@wordpress/i18n';
+import { useState } from '@wordpress/element';
+import { Flex, FlexItem, FlexBlock, CardDivider } from '@wordpress/components';
+import { Spinner } from '@woocommerce/components';
+
+/**
+ * Internal dependencies
+ */
+import useCountryCallingCodes from '.~/hooks/useCountryCallingCodes';
+import useGoogleMCPhoneNumber from '.~/hooks/useGoogleMCPhoneNumber';
+import Section from '.~/wcdl/section';
+import SelectControl from '.~/wcdl/select-control';
+import AppInputControl from '.~/components/app-input-control';
+import AccountCard, { APPEARANCE } from '.~/components/account-card';
+import AppButton from '.~/components/app-button';
+import AppSpinner from '.~/components/app-spinner';
+import './phone-number-card.scss';
+
+const noop = () => {};
+
+function PhoneNumberContent( {
+	initCountry,
+	initNationalNumber,
+	onPhoneNumberChange,
+} ) {
+	const countryCallingCodes = useCountryCallingCodes( 'select-options' );
+	const [ country, setCountry ] = useState( initCountry );
+	const [ number, setNumber ] = useState( initNationalNumber );
+
+	const handleChange = ( nextCountry, nextNumber ) => {
+		if ( nextCountry !== country ) {
+			setCountry( nextCountry );
+		}
+		if ( nextNumber !== number ) {
+			setNumber( nextNumber );
+		}
+
+		const countryCallingCode = nextCountry
+			? getCountryCallingCode( nextCountry )
+			: '';
+		onPhoneNumberChange( countryCallingCode, nextNumber, nextCountry );
+	};
+
+	const handleCountryChange = ( v ) => handleChange( v, number );
+	const handleNumberChange = ( v ) => handleChange( country, v );
```

Comment:
:nail_care: 
- Personally, I try to avoid single-char variable names, as they are less descriptive than full names, harder to `ctrl + f`, sometimes ambiguous.
- Conditions at l.35 and l.38 are know in advance and are de-facto differentiating `handleCountryChange` vs. `handleNumberChange` again. Which is already done.
  WDYT of something like:
  ```js	
	const handlePhoneChange = ( nextCountry, nextNumber ) => {
		const countryCallingCode = nextCountry
			? getCountryCallingCode( nextCountry )
			: '';
		onPhoneNumberChange( countryCallingCode, nextNumber, nextCountry );
	};
	const handleCountryChange = ( nextCountry ) => {
		if ( nextCountry !== country ) {
			setCountry( nextCountry );
		}

		handlePhoneChange( number, nextCountry );
	};
	const handleNumberChange = ( nextNumber ) => {
		if ( nextNumber !== number ) {
			setNumber( nextNumber );
		}

		handlePhoneChange( nextNumber, country );
	};
	```
- [`setState` docs](https://reactjs.org/docs/hooks-reference.html#usestate) state
	> If your update function returns the exact same value as the current state, the subsequent rerender will be skipped completely.
	
	So, if I understand that correctly, `if ( nextSate !== currentState ) {  setState( nextState ); }` is already covered.
	Then we could make it 
	```js
 	const handlePhoneChange = ( nextCountry, nextNumber ) => {
		setCountry( nextCountry );
		setNumber( nextNumber );

		const countryCallingCode = nextCountry
			? getCountryCallingCode( nextCountry )
			: '';
		onPhoneNumberChange( countryCallingCode, nextNumber, nextCountry );
	};

	const handleCountryChange = ( nextCountry ) =>
		handlePhoneChange( nextCountry, number );
	const handleNumberChange = ( nextNumber ) =>
		handlePhoneChange( country, nextNumber );
	```

Reply:
Changed by https://github.com/woocommerce/google-listings-and-ads/pull/917/commits/8f42c78e10794eed924ff97b0c1a671552231880